pkg/config/config.go

```diff
@@ -654,19 +653,50 @@ func (c *config) Load(path string, fsys fs.FS) error {
 		}
 		c.Storage.Buckets[name] = bucket
 	}
-	for slug, function := range c.Functions {
-		// TODO: support configuring alternative entrypoint path, such as index.js
-		if len(function.Entrypoint) == 0 {
-			function.Entrypoint = filepath.Join(builder.FunctionsDir, slug, "index.ts")
-		} else if !filepath.IsAbs(function.Entrypoint) {
-			// Append supabase/ because paths in configs are specified relative to config.toml
-			function.Entrypoint = filepath.Join(builder.SupabaseDirPath, function.Entrypoint)
+
+	// Load functions config
+	pattern := filepath.Join(builder.FunctionsDir, "*", "index.ts")
+	paths, err := fs.Glob(fsys, pattern)
+	if err != nil {
+		return errors.Errorf("failed to glob function slugs: %w", err)
+	}
+
+	if c.Functions == nil {
+		c.Functions = make(FunctionConfig)
+	}
```

Comment:
Globbing functions this way is very fragile because users often have shared components that they don't want to deploy as functions, for eg. `functions/shared/index.ts`.

Reply:
I took the code from https://github.com/supabase/cli/blob/4cf189afa6dc8b9738b1de31e4c06fcb6d5e1aaf/internal/functions/deploy/deploy.go#L49-L66
Should we also change this part then?

Reply:
No need to update deploy for the time being. We left it for backwards compatibility with those who use setup-cli GitHub action.

Users only need to declare if they want to deploy to branches using our new integration.